src/CookieCollection.php

```diff
@@ -0,0 +1,352 @@
+     * @see getValue()
+     */
+    public function get(string $name): Cookie
+    {
+        return $this->cookies[$name];
+    }
+
+    /**
+     * Returns the value of the named cookie.
+     *
+     * @param string $name the cookie name
+     * @param mixed $defaultValue the value that should be returned when the named cookie does not exist.
+     * @return string|null the value of the named cookie or the default value if cookie is not set.
+     * @see get()
+     */
+    public function getValue(string $name, $defaultValue = null): ?string
+    {
+        return isset($this->cookies[$name]) ? $this->cookies[$name]->getValue() : $defaultValue;
+    }
+
+    /**
+     * Adds a cookie to the collection.
+     * If there is already a cookie with the same name in the collection, it will be removed first.
+     *
+     * @param Cookie $cookie the cookie to be added
+     */
+    public function add(Cookie $cookie): void
+    {
+        $this->cookies[$cookie->getName()] = $cookie;
+    }
+
+    /**
+     * Returns whether there is a cookie with the specified name.
+     *
+     * @param string $name the cookie name
+     * @return bool whether the named cookie exists
+     * @see remove()
+     */
+    public function has(string $name): bool
+    {
+        return isset($this->cookies[$name]);
+    }
+
+    /**
+     * Removes a cookie.
+     *
+     * @param string $name the name of the cookie to be removed.
+     * @return Cookie|null cookie that is removed
+     */
+    public function remove(string $name): ?Cookie
+    {
+        if (!isset($this->cookies[$name])) {
+            return null;
+        }
+
+        $removed = $this->cookies[$name];
+        unset($this->cookies[$name]);
+
+        return $removed;
+    }
+
+    /**
+     * Set cookie expire date to outdated to further remove it from browser.
+     *
+     * @param string $name the name of the cookie to expire.
+     */
+    public function expire(string $name): void
+    {
+        if (!isset($this->cookies[$name])) {
+            return;
+        }
+
+        $this->cookies[$name] = $this->cookies[$name]->withExpires(new DateTimeImmutable('-1 year'));
+    }
+
+    /**
+     * Removes all cookies.
+     */
+    public function clear(): void
+    {
+        $this->cookies = [];
+    }
+
+    /**
+     * Returns whether the collection already contains the cookie.
+     *
+     * @param Cookie $cookie the cookie to check for
+     * @return bool whether cookie exists
+     * @see has()
+     */
+    public function contains(Cookie $cookie): bool
+    {
+        return in_array($cookie, $this->cookies, true);
+    }
+
+    /**
+     * Tests for the existence of the cookie that satisfies the given predicate.
+     *
+     * @param Closure $p The predicate.
+     * @return bool whether the predicate is true for at least on cookie.
+     */
+    public function exists(Closure $p): bool
```

Comment:
What's the use case?

Reply:
To check existence of the cookie by some attribute, e.g.:
```php
$collection->exists(fn (string $name, Cookie $cookie) => $cookie->getValue() === 'something');
```

Also an implementation of the planned interface (see answer below).